diff --git a/mangle.c b/mangle.c
index b049168..31447cb 100644
--- a/mangle.c
+++ b/mangle.c
@@ -285,6 +285,21 @@ int tcp4_frag(const __u8 *pkt, __u32 buflen, __u32 payload_offset,
 #define TLS_EXTENSION_SNI 0x0000
 #define TLS_EXTENSION_CLIENT_HELLO_ENCRYPTED 0xfe0d
 
+struct
+{
+	char *domain;
+	int len;
+}
+domains[] =
+{
+	{ "googlevideo.com", 15 },
+	{ "youtube.com", 11 },
+	{ "ggpht.com", 9 },
+	{ "ytimg.com", 9 },
+};
+
+int domains_count = 4;
+
 const char googlevideo_ending[] = "googlevideo.com";
 const int googlevideo_len = 15;
 
@@ -406,14 +421,17 @@ struct verdict analyze_tls_data(
 			vrd.sni_offset = (uint8_t *)sni_name - data;
 			vrd.sni_len = sni_len;
 
-			char *gv_startp = sni_name + sni_len - googlevideo_len;
-			if (sni_len >= googlevideo_len &&
-				sni_len < 128 && 
-				!strncmp(gv_startp, 
-				googlevideo_ending, 
-				googlevideo_len)) {
-
-				vrd.gvideo_hello = 1;
+			for (int i = 0; i < domains_count; ++i) {
+				char *gv_startp = sni_name + sni_len - domains[i].len;
+				if (sni_len >= domains[i].len &&
+					sni_len < 128 && 
+					!strncmp(gv_startp, 
+					domains[i].domain, 
+					domains[i].len))
+				{
+					vrd.gvideo_hello = 1;
+					break;
+				}
 			}
 
 nextExtension:

